include/copc-lib/las/point.hpp

```diff
@@ -1,597 +1,590 @@
     }
 
-    bool ScanDirFlag() const { return point_10_ ? point_10_->ScanDirectionFlag() : point_14_->ScanDirectionFlag(); }
-    void ScanDirFlag(const bool &scan_dir_flag)
+    bool ScanDirectionFlag() const
     {
-        if (point_10_)
-            point_10_->ScanDirectionFlag(scan_dir_flag);
-        else
-            point_14_->ScanDirectionFlag(scan_dir_flag);
+        return extended_point_type_ ? (extended_flags_ >> 6) & 0x1 : (returns_flags_eof_ >> 6) & 0x1;
     }
-
-    bool EdgeOfFlightLineFlag() const
+    void ScanDirectionFlag(const bool &scan_direction_flag)
     {
-        return point_10_ ? point_10_->EdgeOfFlightLineFlag() : point_14_->EdgeOfFlightLineFlag();
+        if (extended_point_type_)
+            extended_flags_ = (extended_flags_ & 0xBF) | (scan_direction_flag << 6);
+        else
+            returns_flags_eof_ = (returns_flags_eof_ & 0xBF) | (scan_direction_flag << 6);
     }
-    void EdgeOfFlightLineFlag(const bool &eofl_flag)
+
+    bool EdgeOfFlightLineFlag() const { return extended_point_type_ ? extended_flags_ >> 7 : returns_flags_eof_ >> 7; }
+    void EdgeOfFlightLineFlag(const bool &edge_of_flight_line)
     {
-        if (point_10_)
-            point_10_->EdgeOfFlightLineFlag(eofl_flag);
+        if (extended_point_type_)
+            extended_flags_ = (extended_flags_ & 0x7F) | (edge_of_flight_line << 7);
         else
-            point_14_->EdgeOfFlightLineFlag(eofl_flag);
+            returns_flags_eof_ = (returns_flags_eof_ & 0x7F) | (edge_of_flight_line << 7);
     }
 
     uint8_t ClassificationBitFields() const
     {
-        if (point_10_)
-            return point_10_->ClassificationBitFields();
-        else
+        if (extended_point_type_)
             throw std::runtime_error("Point14 does not have Classification BitFields.");
+        else
+            return classification_;
     }
 
     void ClassificationBitFields(const uint8_t &classification_bitfields)
     {
-        if (point_10_)
-            point_10_->ClassificationBitFields(classification_bitfields);
-        else
+        if (extended_point_type_)
             throw std::runtime_error("Point14 does not have Classification Bit Fields.");
+        else
+            classification_ = classification_bitfields;
     }
 
-    uint8_t Classification() const { return point_10_ ? point_10_->Classification() : point_14_->Classification(); }
+    uint8_t Classification() const { return extended_point_type_ ? extended_classification_ : classification_ & 0x1F; }
     void Classification(const uint8_t &classification)
     {
-        if (point_10_)
-            point_10_->Classification(classification);
+        if (extended_point_type_)
+            extended_classification_ = classification;
         else
-            point_14_->Classification(classification);
+        {
+            if (classification > 31)
+                throw std::runtime_error("Classification for Point10 must be <= 31");
```

Comment:
Add a note to the exception that if you want to change the point10 raw classification (or want to override this check), use ClassificationBitField

Reply:
It can't be over 31 though since there are only 5 bits available in the bitfield?

Reply:
Yes but technically you can set all 8 bits. Some viewers still show that as a "classification", such as the ones that don't support flags. So sometimes you want to treat it as an 8 bit field. It might be good just to make a note of it.

Reply:
ok